File: src/bin/mt/apksign/ByteArrayUtil.java
package bin.mt.apksign;

class ByteArrayUtil {

    static void setInt(int value, byte[] data, int offset) {
        data[offset] = (byte) (value & 0xff);
        data[offset + 1] = (byte) ((value >> 8) & 0xff);
        data[offset + 2] = (byte) ((value >> 16) & 0xff);
        data[offset + 3] = (byte) ((value >> 24) & 0xff);
    }

    static void setUInt(long value, byte[] data, int offset) {
        data[offset] = (byte) (value & 0xff);
        data[offset + 1] = (byte) ((value >> 8) & 0xff);
        data[offset + 2] = (byte) ((value >> 16) & 0xff);
        data[offset + 3] = (byte) ((value >> 24) & 0xff);
    }

    static void setLong(long value, byte[] data, int offset) {
        data[offset] = (byte) (value & 0xff);
        data[offset + 1] = (byte) ((value >> 8) & 0xff);
        data[offset + 2] = (byte) ((value >> 16) & 0xff);
        data[offset + 3] = (byte) ((value >> 24) & 0xff);
        data[offset + 4] = (byte) ((value >> 32) & 0xff);
        data[offset + 5] = (byte) ((value >> 40) & 0xff);
        data[offset + 6] = (byte) ((value >> 48) & 0xff);
        data[offset + 7] = (byte) ((value >> 56) & 0xff);
    }

    static long readUInt(byte[] data, int offset) {
        long ch1 = data[offset] & 0xffL;
        long ch2 = data[offset + 1] & 0xffL;
        long ch3 = data[offset + 2] & 0xffL;
        long ch4 = data[offset + 3] & 0xffL;
        return (ch1) | (ch2 << 8) | (ch3 << 16) | (ch4 << 24);
    }

    static byte[] intToBytes(int value) {
        byte[] array = new byte[4];
        setInt(value, array, 0);
        return array;
    }

}

File: Android/app/src/main/java/bin/mt/apksign/ByteArrayUtil.kt
package bin.mt.apksign

internal object ByteArrayUtil {
    fun setInt(value: Int, data: ByteArray, offset: Int) {
        data[offset] = (value and 0xff).toByte()
        data[offset + 1] = (value shr 8 and 0xff).toByte()
        data[offset + 2] = (value shr 16 and 0xff).toByte()
        data[offset + 3] = (value shr 24 and 0xff).toByte()
    }

    fun setUInt(value: Long, data: ByteArray, offset: Int) {
        data[offset] = (value and 0xffL).toByte()
        data[offset + 1] = (value shr 8 and 0xffL).toByte()
        data[offset + 2] = (value shr 16 and 0xffL).toByte()
        data[offset + 3] = (value shr 24 and 0xffL).toByte()
    }

    fun setLong(value: Long, data: ByteArray, offset: Int) {
        data[offset] = (value and 0xffL).toByte()
        data[offset + 1] = (value shr 8 and 0xffL).toByte()
        data[offset + 2] = (value shr 16 and 0xffL).toByte()
        data[offset + 3] = (value shr 24 and 0xffL).toByte()
        data[offset + 4] = (value shr 32 and 0xffL).toByte()
        data[offset + 5] = (value shr 40 and 0xffL).toByte()
        data[offset + 6] = (value shr 48 and 0xffL).toByte()
        data[offset + 7] = (value shr 56 and 0xffL).toByte()
    }

    fun readUInt(data: ByteArray, offset: Int): Long {
        val ch1 = data[offset].toLong() and 0xffL
        val ch2 = data[offset + 1].toLong() and 0xffL
        val ch3 = data[offset + 2].toLong() and 0xffL
        val ch4 = data[offset + 3].toLong() and 0xffL
        return ch1 or (ch2 shl 8) or (ch3 shl 16) or (ch4 shl 24)
    }

    fun intToBytes(value: Int): ByteArray {
        val array = ByteArray(4)
        setInt(value, array, 0)
        return array
    }
}